zcash_test_vectors/unified_address.py

```diff
@@ -21,21 +22,29 @@ def main():
 
     rng = Random(0xabad533d)
     rand = Rand(randbytes(rng))
+    seed = bytes(range(32))
 
     test_vectors = []
-    for _ in range(0, 10):
+    for account in range(0, 10):
         has_t_addr = rand.bool()
         if has_t_addr:
             t_addr = b"".join([rand.b(20)])
         else:
             t_addr = None
 
+        j = 0
         has_s_addr = rand.bool()
         if has_s_addr:
-            sapling_sk = sapling_key_components.SpendingKey(rand.b(32))
-            sapling_default_d = sapling_sk.default_d()
-            sapling_default_pk_d = sapling_sk.default_pkd()
-            sapling_raw_addr = b"".join([sapling_default_d[:11], bytes(sapling_default_pk_d)[:32]])
+            rand.b(32) # discard
```

Comment:
The commits are written in the following order:
1. trivial refactor
2. changes only the Sapling vectors
3. changes only the Orchard vectors
4. changes only the transparent vectors
5. cleanup that ensures we are using the randomness in precisely the same way for unified_address, unified_incoming_viewing_keys, and unified_full_viewing_keys (so corresponding test vectors are about the same keys)
6. optimization

In commits 2, 3, and 4, we're careful to retain all of the sampling of randomness, even if it is now discarded (because otherwise all of the test vectors after that point would change). This is because @str4d wanted to be able to verify commit-by-commit that only the expected vectors had changed. However, then I realised that making the test vectors "match up" between unified_incoming_viewing_keys, and unified_full_viewing_keys would change all the vectors at once in commit 5, anyway. But I still think this approach made it easier for me to check that the vectors were only changing in expected ways (even if it took a long time to get it right).